src/Data/List/Sufficient.agda

```diff
@@ -0,0 +1,57 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- 'Sufficient' lists: a structurally inductive view of lists xs
+-- as given by xs ≡ non-empty prefix + sufficient suffix
+--
+-- Useful for termination arguments for function definitions
+-- which provably consume a non-empty (but otherwise arbitrary) prefix
+-- *without* having to resort to ancillary WF induction on length etc.
+-- e.g. lexers, parsers etc.
+--
+-- Credited by Conor McBride as originally due to James McKinna
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.List.Sufficient {a} {A : Set a} where
+
+open import Data.List.Base using (List; []; _∷_; _++_)
+open import Relation.Binary.PropositionalEquality using (_≡_; refl)
+
+
+------------------------------------------------------------------------
+-- Sufficient view
+
+data Sufficient : (xs : List A) → Set a where
+
+  suff-acc : ∀ {xs} →
+             (suff_ih : ∀ {a} {prefix suffix} → xs ≡ a ∷ prefix ++ suffix →
+                        Sufficient suffix) →
+             Sufficient xs
+
+
+------------------------------------------------------------------------
+-- Sufficient properties: constructors (typically not for export)
```

Comment:
So make them `private`?

Reply:
So... rather like the discussion if other recent work, I supposed there might be cases where clients would want access to them; so I guess thus represents a... hesitation on my part.

Reply:
Further to that: I realised that I didn't know how to mark the constructors off as being '*likely* to be private/hidden during `import`', but without committing to that choice, and then finding that I needed immediately (!)  to use them in the scope of `sufficient`... so perhaps I don't need the explicit `module _ where` around *that* definition? 

is it the case that the `open Constructors` *won't*  export them from the module (whereas `open ... public` *would*)? I confess I still trip up over this in Agda... :-(

Reply:
OK, but as below (comments maybe got put in wrong place), I'd rather leave `Constructors` as *solely* relevant to constructing the view covering function `sufficient`.